src/main/scala/com/microsoft/azure/sqldb/spark/sql/streaming/SQLWriter.scala

```diff
@@ -0,0 +1,193 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.azure.sqldb.spark.sql.streaming
+
+import java.sql._
+
+import com.microsoft.azure.sqldb.spark.config.{Config, SqlDBConfig}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.sql.{DataFrame, SaveMode, SparkSession}
+import org.apache.spark.sql.catalyst.InternalRow
+import com.microsoft.azure.sqldb.spark.connect.ConnectionUtils._
+import org.apache.spark.sql.catalyst.expressions._
+
+import scala.Array
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/**
+  * The [[SQLWriter]] class is used to write data from a batch query
+  * or structured streaming query, given by a [[QueryExecution]], to Azure SQL Database or Azure SQL Data Warehouse.
+  */
+private[spark] object SQLWriter extends Logging {
+
+  var subset = false
+  var DRIVER_NAME: String = "com.microsoft.sqlserver.jdbc.SQLServerDriver"
+  var connection: Connection = _
+  val daysToMillis = 24L*60L*60L*1000L
+
+  override def toString: String = "SQLWriter"
+
+  def write(
+             sparkSession: SparkSession,
+             data: DataFrame,
+             queryExecution: QueryExecution,
+             saveMode: SaveMode,
+             parameters: Map[String, String]
+           ): Unit = {
+
+    //TODO: (Future Release) Provide the option to the user to define the columns they're writing to and/or column mapping
+    var writeConfig = Config(parameters)
+    val url = writeConfig.get[String](SqlDBConfig.URL).getOrElse(
+      throw new IllegalArgumentException("URL not found in Config")
+    )
+    val db = writeConfig.get[String](SqlDBConfig.DatabaseName).getOrElse(
+      throw new IllegalArgumentException("Database not found in Config")
+    )
+    val table = writeConfig.get[String](SqlDBConfig.DBTable).getOrElse(
+      throw new IllegalArgumentException("table not found in Config")
+    )
+    val user = writeConfig.get[String](SqlDBConfig.User).getOrElse(
+      throw new IllegalArgumentException("user not found in Config")
+    )
+    val password = writeConfig.get[String](SqlDBConfig.Password).getOrElse(
+      throw new IllegalArgumentException("password not found in Config")
+    )
+    val port = writeConfig.get[String](SqlDBConfig.PortNumber).getOrElse("1433")
+    val ignoreColumnNames = Try(writeConfig.get[String]("ignoreColumnNames").map(_.toBoolean).getOrElse(false)).getOrElse(false)
+
+    //Using regular writes
+    Class.forName(DRIVER_NAME)
+    connection = DriverManager.getConnection(createJDBCUrl(url, Some(port)) + ";database=" + db, user, password)
```

Comment:
Please reuse the connection mechanism used in existing code paths, so that other connection options like AAD authentication etc. can be used. The current implementation seems to impose only user + password authentication which is limiting for some users who want to use AAD.